Pal/src/host/Linux-SGX/sgx_framework.c

```diff
@@ -297,6 +297,27 @@ int add_pages_to_enclave(sgx_arch_secs_t * secs,
     return 0;
 }
 
+/* mktcs sends IOCTL to ask sgx driver to change 
+ * the type of a regular page to TCS type */
+void mktcs(unsigned long tcs_addr)
+{
+#if SDK_DRIVER_VERSION == KERNEL_VERSION(2, 0, 0)
+    struct sgx_range params;
+    memset(&params, 0 ,sizeof(struct sgx_range));
+    params.start_addr = tcs_addr;
+    params.nr_pages = 1;
+    int ret = 0;
+
+    ret = INLINE_SYSCALL(ioctl, 3, isgx_device, SGX_IOC_ENCLAVE_MKTCS, &params);
+    if (IS_ERR(ret)) {
+	SGX_DBG(DBG_I, "Enclave MKTCS returned %d\n", ret);
+	return ;
+    }
+#else
+    SGX_DBG(DBG_E, "EDMM is not supported by SDK before 2.0\n");
```

Comment:
I think that's not SDK, it's SGX driver?  https://github.com/intel/linux-sgx-driver/blob/master/sgx_ioctl.c#L383.

Reply:
@duanbing Yes, you are correct. But this PR is 3 years old and not planned for merging. We keep this PR open simply to not forget about a new version of the EDMM support :)

Reply:
@vijaydhanraj is reviving these efforts, if you're interested then please take a look at #2190.